train.py

```diff
@@ -10,12 +11,16 @@
 import torch.nn.functional as F
 from torch.nn.utils import clip_grad_norm_
 from torch.utils.data import DataLoader
+import torch.distributed as dist
+from torch.distributed._composable.fsdp import fully_shard, MixedPrecisionPolicy
+
 from transformers import (
     AutoTokenizer,
     PreTrainedTokenizer,
     LlamaForCausalLM,
     GenerationConfig,
 )
+from ckpt_utils import save_checkpoint
```

Comment:
;-) you are introducing the utils file again which I removed for simplicity .. intuitively I would also put that in a separate file, let's do this.

Reply:
Do you mean having simply a `utils.py` file instead of the `ckpt_utils.py`? 

Unfortunately the torch distributes save/load are quite verbose so IMO better to move it outside the train code but really up to you